### PR TITLE
Add Storm runas command (SYN-2872)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/storm.py [hunk 1]
```diff
@@ -4530,3 +4530,69 @@ async def execStormCmd(self, runt, genr):
...
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('user', help='The user name or iden to execute the storm query as.')
+        pars.add_argument('storm', help='The storm query to execute.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        if runt.user is not None and not runt.user.isAdmin():
```
Comment (invisig0th) [inline synapse/lib/storm.py:4556]: runt.user should not ever be None, maybe we should just deref it and blow up if it is?

File: synapse/lib/storm.py [hunk 2]
```diff
@@ -4530,3 +4530,69 @@ async def execStormCmd(self, runt, genr):
...
+
+            query = await runt.getStormQuery(text)
+            user = await core.auth.reqUserByNameOrIden(user)
+
+            opts = {'user': user}
+
+            async with await core.snap(user=user, view=runt.snap.view) as snap:
+                async with await Runtime.anit(query, snap, user=user, opts=opts, root=runt) as subr:
+                    subr.debug = runt.debug
+                    subr.readonly = runt.readonly
```
Comment (invisig0th) [inline synapse/lib/storm.py:4595]: Did we make a conscious decision not to propagate asroot here?

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/storm.py (hunk 2 above)
Comment (Cisphyx) [reply synapse/lib/storm.py:4595]: I was assuming we wanted to use the user's permissions when executing the query

Comment (Cisphyx) [reply synapse/lib/storm.py:4599]: added an `--asroot` arg to control propagating asroot.

**Review by invisig0th — APPROVED** [review]

**Pull request merged**